client/lib/save-form/index.js

```diff
@@ -34,7 +33,7 @@ const saveForm = ( setIsSaving, setSuccess, setError, url, nonce, formData ) =>
 			}
 
 			return setError( JSON.stringify( json ) );
-		} );
+		} ).then( () => setIsSaving( false ) );
```

Comment:
To make sure that `setIsSaving(false)` is the last thing this function does. I use it in the `nextStep` action to detect when the form finished validating.